src/CoxeterMatrices.jl

```diff
@@ -0,0 +1,216 @@
+export coxeter_matrix_from_group_type
+
+"""
+    coxeter_matrix_from_group_type(groupType::String)
+
+Create a coxeter matrix for a given group type. Affine groups are denoted by including a '~' character
+after a letter.
+
+#Examples
+```julia-repl
+julia> coxeter_matrix_from_group_type("A4")
+4×4 Matrix{Int64}:
+ 1  3  2  2
+ 3  1  3  2
+ 2  3  1  3
+ 2  2  3  1
+```
+```julia-repl
+julia> coxeter_matrix_from_group_type("D~4")
+5×5 Matrix{Int64}:
+ 1  2  3  2  2
+ 2  1  3  2  2
+ 3  3  1  3  3
+ 2  2  3  1  2
+ 2  2  3  2  1
+
+```
+"""
+function coxeter_matrix_from_group_type(groupType::String)
+    parsedGroupType = validate_and_parse_coxeter_group_type_string(groupType)
+    return build_coxeter_matrix_from_group_type(parsedGroupType[1], parsedGroupType[2], parsedGroupType[3])
+end
+
+"""
+    build_coxeter_matrix_from_group_type(C::Char, N::Int, isAffine::Bool)
+
+Take a given group type letter (A through I), group type integer, and boolean flag if the group is affine. Return the 
+corresponding Coxeter Matrix, if it is possible to construct.
+
+Matrices are constructed based on the diagrams in Kac's Infinite Dimensional Lie Algebras, pg. 43
+"""
+function build_coxeter_matrix_from_group_type(C::Char, N::Int, isAffine::Bool)
+    if C == 'A'
+        if !isAffine
+            M = fill(2,N,N)
+            for i = 1:N
+                M[i,i] = 1
+                if i != N
+                    M[i,i+1] = 3
+                    M[i+1,i] = 3
+                end
+            end
+        else
+            M = build_coxeter_matrix_from_group_type('A',N+1,false)
```

Comment:
`A~1` is a special case, equal to `I2(∞)`. And while it makes sense to define `A0` (definitely keep this), `A~0` does not make sense so add a check for `N ≥ 1` there.